### ISSUE TITLE
tss2-esapi

### ISSUE THREAD

**Issue by junmo-weiyi** [tpm2-software/tpm2-tss#2866]
When I tried to use anaconda to install tpm2-pytss, a fault has occured.
"raise RuntimeError("Require tss2-esapi to be installed and at least version 2.4.0")
      RuntimeError: Require tss2-esapi to be installed and at least version 2.4.0
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed"
But I have found all foulder and no result. 
And I want to know,when I used vs2019 to produse solution , It also didn't have tss2-esapi. 
How can I do 
Please help

**Comment by joholl** [comment]
TL:DR; install `libtss2-dev`.

## Long version

My guess: you have the wrong package installed. Python should identify the library via `pkg-config`:

```sh
pkg-config --modversion tss2-esys
```

`pkg-config` will only find your library if there is a file `tss2-esys.pc` in a location where `pkg-config` actually looks. In my case, it was in `/usr/lib/x86_64-linux-gnu/pkgconfig/tss2-esys.pc`.

For Ubuntu Jammy this file is not distributed via the `libtss2-esys` package (see [file list](https://packages.ubuntu.com/jammy/amd64/libtss2-esys-3.0.2-0/filelist)), but you have to install the development version of it: `libtss2-dev` (see [file list](https://packages.ubuntu.com/jammy/amd64/libtss2-dev/filelist)).
